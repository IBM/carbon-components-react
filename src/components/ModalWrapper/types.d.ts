declare module 'carbon-components-react' {
    import {Component} from 'react';

    export type ModalWrapperTriggerButtonKind = 'primary' | 'secondary' | 'danger' | 'ghost';

    export interface ModalWrapperProps {
        status?: string;
        handleOpen?: (...args: any[])=>any;
        children?: React.ReactNode;
        id?: string;
        buttonTriggerText?: string;
        modalLabel?: string;
        modalHeading?: string;
        modalText?: string;
        passiveModal?: boolean;
        withHeader?: boolean;
        modalBeforeContent?: boolean;
        primaryButtonText?: string;
        secondaryButtonText?: string;
        handleSubmit?: (...args: any[])=>any;
        disabled?: boolean;
        triggerButtonKind?: ModalWrapperTriggerButtonKind;
        shouldCloseAfterSubmit?: boolean;
    }

    export default class ModalWrapper extends Component<ModalWrapperProps, any> {
        render(): JSX.Element;

    }

}

