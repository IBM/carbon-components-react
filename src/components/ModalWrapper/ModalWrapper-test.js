import React from 'react';
import { mount } from 'enzyme';
import ModalWrapper from '../ModalWrapper';

describe('ModalWrapper', () => {
  let mockProps = {};

  beforeEach(() => {
    mockProps = {
      id: 'modal',
      buttonTriggerText: 'Test Modal',
      modalHeading: 'Transactional Modal',
      modalLabel: 'Test Modal Label',
      primaryButtonText: 'Save',
      secondaryButtonText: 'Cancel',
      handleSubmit: jest.fn(() => true),
      shouldCloseAfterSubmit: true,
    };
  });

  it('should render', () => {
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the open state upon change in props', () => {
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );

    wrapper.setProps({ open: true });
    expect(wrapper.state('isOpen')).toBe(true);

    wrapper.setProps({ open: false });
    expect(wrapper.state('isOpen')).toBe(false);
  });

  it('should avoid changing the open state when receiving new props unless the value has changed', () => {
    const wrapper = mount(
      <ModalWrapper {...mockProps} open={true}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );

    wrapper.find({ children: mockProps.primaryButtonText }).simulate('click'); // Turns `state.open` to `false`
    wrapper.setProps({ open: true }); // No change in `open` prop
    expect(wrapper.state('isOpen')).toBe(false);
  });

  it('should close after a successful submit action', () => {
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );
    wrapper.find({ children: mockProps.buttonTriggerText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);

    wrapper.find({ children: mockProps.primaryButtonText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(false);
  });

  it('should not close after an unsuccessful submit action', () => {
    mockProps.handleSubmit = jest.fn(() => false);
    const wrapper = mount(
      <ModalWrapper {...mockProps}>
        <p className="bx--modal-content__text">Text</p>
      </ModalWrapper>
    );
    wrapper.find({ children: mockProps.buttonTriggerText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);

    wrapper.find({ children: mockProps.primaryButtonText }).simulate('click');
    expect(wrapper.state('isOpen')).toBe(true);
  });
});
