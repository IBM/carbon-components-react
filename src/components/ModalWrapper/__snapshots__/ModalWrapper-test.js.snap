// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper should render 1`] = `
<ModalWrapper
  buttonTriggerClassName="btn-trigger"
  buttonTriggerText="Test Modal"
  disabled={false}
  handleSubmit={[MockFunction]}
  id="modal"
  modalHeading="Transactional Modal"
  modalLabel="Test Modal Label"
  onKeyDown={[Function]}
  primaryButtonText="Save"
  secondaryButtonText="Cancel"
  selectorPrimaryFocus="[data-modal-primary-focus]"
  shouldCloseAfterSubmit={true}
  triggerButtonIconDescription="Provide icon description if icon is used"
  triggerButtonKind="primary"
>
  <div
    onKeyDown={[Function]}
    role="presentation"
  >
    <ForwardRef
      className="btn-trigger"
      disabled={false}
      iconDescription="Provide icon description if icon is used"
      kind="primary"
      onClick={[Function]}
      small={false}
      tabIndex={0}
      type="button"
    >
      <button
        className="btn-trigger bx--btn bx--btn--primary"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="button"
      >
        Test Modal
      </button>
    </ForwardRef>
    <Modal
      focusTrap={true}
      iconDescription="close the modal"
      id="modal"
      modalHeading="Transactional Modal"
      modalLabel="Test Modal Label"
      onKeyDown={[Function]}
      onRequestClose={[Function]}
      onRequestSubmit={[Function]}
      open={false}
      passiveModal={false}
      primaryButtonDisabled={false}
      primaryButtonText="Save"
      secondaryButtonText="Cancel"
      selectorPrimaryFocus="[data-modal-primary-focus]"
    >
      <FocusTrap
        _createFocusTrap={[Function]}
        active={false}
        focusTrapOptions={Object {}}
        paused={false}
      >
        <div
          className="bx--modal bx--modal-tall"
          id="modal"
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="presentation"
          tabIndex={-1}
        >
          <div
            aria-modal="true"
            className="bx--modal-container"
            role="dialog"
          >
            <div
              className="bx--modal-header"
            >
<<<<<<< HEAD
              <Close20
                aria-label="close the modal"
                className="bx--modal-close__icon"
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <svg
                  aria-label="close the modal"
                  className="bx--modal-close__icon"
                  focusable="false"
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
                  />
                </svg>
              </Close20>
            </button>
          </div>
          <div
            className="bx--modal-content"
          >
            <p
              className="bx--modal-content__text"
=======
              <p
                className="bx--modal-header__label"
              >
                Test Modal Label
              </p>
              <p
                className="bx--modal-header__heading"
              >
                Transactional Modal
              </p>
              <button
                className="bx--modal-close"
                onClick={[Function]}
                title="close the modal"
                type="button"
              >
                <Icon
                  className="bx--modal-close__icon"
                  description="close the modal"
                  fillRule="evenodd"
                  icon={
                    Object {
                      "height": "10",
                      "id": "icon--close",
                      "name": "icon--close",
                      "styles": "",
                      "svgData": Object {
                        "circles": "",
                        "ellipses": "",
                        "paths": Array [
                          Object {
                            "d": "M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z",
                            "fill-rule": "nonzero",
                          },
                        ],
                        "polygons": "",
                        "polylines": "",
                        "rects": "",
                      },
                      "tags": "icon--close",
                      "viewBox": "0 0 10 10",
                      "width": "10",
                    }
                  }
                  role="img"
                >
                  <svg
                    alt="close the modal"
                    aria-label="close the modal"
                    className="bx--modal-close__icon"
                    fillRule="evenodd"
                    height="10"
                    role="img"
                    viewBox="0 0 10 10"
                    width="10"
                  >
                    <title>
                      close the modal
                    </title>
                    <path
                      d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
                      key="key0"
                    />
                  </svg>
                </Icon>
              </button>
            </div>
            <div
              className="bx--modal-content"
>>>>>>> b9f5d9e92dd81dce104dea313a7de9989de9910c
            >
              <p
                className="bx--modal-content__text"
              >
                Text
              </p>
            </div>
            <div
              className="bx--modal-footer"
            >
              <ForwardRef
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                kind="secondary"
                onClick={[Function]}
                small={false}
                tabIndex={0}
                type="button"
              >
<<<<<<< HEAD
                Cancel
              </button>
            </ForwardRef>
            <ForwardRef
              disabled={false}
              iconDescription="Provide icon description if icon is used"
              inputref={
                Object {
                  "current": <button
                    class="bx--modal-close"
                    title="close the modal"
                    type="button"
                  >
                    <svg
                      aria-label="close the modal"
                      class="bx--modal-close__icon"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
                      />
                    </svg>
                  </button>,
                }
              }
              kind="primary"
              onClick={[Function]}
              small={false}
              tabIndex={0}
              type="button"
            >
              <button
                className="bx--btn bx--btn--primary"
=======
                <button
                  className="bx--btn bx--btn--secondary"
                  disabled={false}
                  onClick={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Cancel
                </button>
              </ForwardRef>
              <ForwardRef
>>>>>>> b9f5d9e92dd81dce104dea313a7de9989de9910c
                disabled={false}
                iconDescription="Provide icon description if icon is used"
                inputref={
                  Object {
                    "current": <button
                      class="bx--btn bx--btn--primary"
                      tabindex="0"
                      type="button"
                    >
                      Save
                    </button>,
                  }
                }
                kind="primary"
                onClick={[Function]}
                small={false}
                tabIndex={0}
                type="button"
              >
                <button
                  className="bx--btn bx--btn--primary"
                  disabled={false}
                  onClick={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Save
                </button>
              </ForwardRef>
            </div>
          </div>
        </div>
      </FocusTrap>
    </Modal>
  </div>
</ModalWrapper>
`;
